slam/src/slam/robot_physics/RobotSimulatorPhysics.py

```diff
@@ -0,0 +1,68 @@
+import copy
+from slam.fast_slam2.FastSLAM2 import FastSLAM2
+from slam.robot_physics.RobotPhysics2D import RobotPhysics2D
+import numpy as np
+from slam.Util import wrap_to_pi
+
+class RobotPhysics2DForSim(RobotPhysics2D):
```

Comment:
Again, a few comments here and there (esp explaining dense math) would help. But I really like this new class structure.